app/controllers/edit_directory_columns_controller.rb

```diff
@@ -26,16 +24,33 @@ def update
       raise Discourse::InvalidParameters, "Must have at least one column enabled"
     end
 
+    new_values = ""
+    previous_values = ""
+    staff_action_logger = StaffActionLogger.new(current_user)
+
     directory_column_params[:directory_columns].values.each do |column_data|
       existing_column = directory_columns.detect { |c| c.id == column_data[:id].to_i }
       if (
-           existing_column.enabled != column_data[:enabled] ||
+           existing_column.enabled != ActiveModel::Type::Boolean.new.cast(column_data[:enabled]) ||
              existing_column.position != column_data[:position].to_i
          )
-        existing_column.update(enabled: column_data[:enabled], position: column_data[:position])
+        new_value, previous_value =
+          staff_action_logger.edit_directory_columns_details(column_data, existing_column)
+
+        new_values += new_value
+        previous_values += previous_value
+
+        existing_column.update(
+          enabled: column_data[:enabled],
+          position: column_data[:position].to_i,
+        )
       end
     end
 
+    details = {}
+
+    details.merge!({ previous_value: previous_values, new_value: new_values })
+    staff_action_logger.log_custom("update_directory_columns", details = details)
```

Comment:
```suggestion
    staff_action_logger.log_custom("update_directory_columns", { previous_value: previous_values, new_value: new_values })
```

Would this work too?